modules/data.atmosphere/R/met_temporal_downscale.Gaussian_ensemble.R

```diff
@@ -0,0 +1,398 @@
+              lowday <- x + n
+              highday <- x + n + wd * sd_step + n
+            }
+            if (highday > reso_len) {
+              highday <- reso_len
+              lowday <- x * div - wd * sd_step + n
+            }
+            four[n] <- rnorm(1, mean = sour[x], sd = sd(a[seq(from = lowday, 
+                                                              to = highday, by = sd_step)]))
+          }
+          woo <- append(woo, four)
+        }
+      }
+      df[1:length(woo), u] <- woo
+    }
+    
+    # Makes sure that max temps >= avg temp and min temps <= avg temp
+    for (x in seq_along(df$air_temperature)) {
+      if (all(is.na(df$air_temperature_max)) == FALSE) {
+        if (df$air_temperature_max[x] < df$air_temperature[x]) {
+          df$air_temperature_max[x] <- df$air_temperature[x]
+        }
+      }
+      if (all(is.na(df$air_temperature_max)) == TRUE) {
+        df$air_temperature_max <- rep(NA, reso_len)
+      }
+      if (all(is.na(df$air_temperature_min)) == FALSE) {
+        if (df$air_temperature_min[x] > df$air_temperature[x]) {
+          df$air_temperature_min[x] <- df$air_temperature[x]
+        }
+      }
+      if (all(is.na(df$air_temperature_min)) == TRUE) {
+        df$air_temperature_min <- rep(NA, reso_len)
+      }
+    }
+    
+    # Precipitation_flux this takes the daily total of precipitation and uses that as
+    # a total possible amount of precip.  It randomly distributes the values of
+    # precipitation
+    rand_vect_cont <- function(N, M, sd = 1) {
+      vec <- rnorm(N, M/N, sd)
+      vec/sum(vec) * M
+    }
+    precip <- vector()
+    for (x in seq_along(source$precipitation_flux)) {
+      lowday <- (x - wd) * div
+      highday <- (x + wd) * div
+      if (lowday < 0) {
+        lowday <- 0
+      }
+      if (highday > reso_len) {
+        highday <- reso_len
+      }
+      four <- vector()
+      four <- rand_vect_cont(div, source$precipitation_flux[x], sd = sd(train$precipitation_flux[lowday:highday]))
+      four[four < 0] <- 0
+      precip <- append(precip, four)
+    }
+    df$precipitation_flux <- precip
+    
+    # Specific Humidity, eastward wind and northward wind
+    wnd <- c("specific_humidity", "eastward_wind", "northward_wind", "surface_downwelling_longwave_flux_in_air", 
+             "air_pressure")
+    for (u in wnd) {
+      woo <- vector()
+      a <- as.numeric(train[[u]])
+      sour <- as.numeric(source[[u]])
+      if (all(is.na(sour)) == TRUE) {
+        woo <- rep(NA, reso_len)
+      }
+      if (all(is.na(sour)) == FALSE) {
+        for (x in seq_along(sour)) {
+          lowday <- (x - wd) * div
+          highday <- (x + wd) * div
+          if (lowday < 0) {
+            lowday <- 0
+          }
+          if (highday > reso_len) {
+            highday <- reso_len
+          }
+          four <- vector()
+          for (n in seq_along(1:div)) {
+            four[n] <- rnorm(1, mean = sour[x], sd = sd(a[lowday:highday]))
```

Comment:
In the conditional version, you have choices on what you condition on. For example, hydrologists often start by conditioning on the downscaled precip. Christy started with air T and then conditioned other variables on that.

One simple way to do that is to just filter the values in your window -- e.g. if you start with precip, once you draw the precip only use the values in your window that have a similar precip to choose from.  This is similar to how the MDS gap-filling code works. Indeed, you might want to look at the gapfilling lit as well because downscaling is very similar to gap-filling, it's just that the gaps are very regular and persistent.